### DISCUSSION TITLE
Install image on specific partition instead on entire disk

### DISCUSSION THREAD

**Discussion by pugant** [Joshua-Riek/ubuntu-rockchip#837] [General]
Good morning,
in reality this is not an issue, but more a request for information and any documentation.
For future updates that come out, I would like to setup my NVME device on Orange Pi 5 Plus with two partitions, so I can put the OS on one and reserve the other for data.
This way I will be able to update only the operating system partition with each release and not destroy the entire device with each update.

I tried to do this by partitioning my disk with two partitions:
- nvme0n1p1 (Operating System)
- nvme0n1p2 (Data)

I then flashed the image on the nvme0n1p1 partition with dd and rebooted, but what happens is that upon reboot I get the Orange splash screen and no system loads
I would like to know if this is possible and if so, what are the steps (with lots of detail) to be able to do it.

**Comment by hakotani-o** [comment]
Hello.
I don't have an NVME DISK, so I can't confirm, but I was able to resize a USB-connected DISK.

Is it possible to boot ubuntu desktop from SD-CARD?

After booting,
There is an application called DISK in Utilities, so start it.

If NVME DISK is displayed, I think you can resize it.

Please note that if you use the same disk image for SD-CARD and NVME DISK, the UUIDs will overlap.

I wish you the best.

**Comment by pugant** [comment]
Hello,
I tried to resize the partitions successfully, directly on my nvme disk after installing the image using dd and I succeeded without problems.
Unfortunately however, when we will have the next version of the image (for example version 3.0) thanks to the fantastic work of Joshua, if I want to update to the new version I will have to destroy the partition scheme and flash the image on the entire disk instead of on the specific partition .
If someone already knows how it is possible to do it and wants to provide step-by-step guidance, I believe they will make an important contribution to the correct management and updating of their SBC

**Comment by robalexj** [comment]
I too would find this useful. I'd be seeking to partition it in a similar way to pugant, and then have the ability to re-install Ubuntu without wiping any separate data partitions.

**Comment by DiegoBM** [comment]
Does this distro not allow to do a dist upgrade via apt? I reckon you can grab Joshua's updates via apt, and I presume you wouldn't need to reflash if you upgrade that way once you have resized the OS parition.

**Reply by FreeWilly2022** [reply]
Yes, this distro updates via apt, its on the main readme "Receive all updates and changes through apt".
